charts/astronomer/templates/commander/commander-role.yaml

```diff
@@ -1,19 +1,33 @@
 ########################################
 ## Astronomer Commander Role/ClusterRole
 ########################################
-{{- $singleNamespace := .Values.global.singleNamespace }}
-{{- $clusterRoles:= .Values.global.clusterRoles }}
-{{- if and .Values.global.rbacEnabled }}
-kind: {{ if or ($singleNamespace) (not $clusterRoles) }}Role{{ else }}ClusterRole{{ end }}
+# Here, we either:
+# 1. Do not create any resource if rbacEnabled is disabled
+# 2. Create a Cluster Role if clusterRoles and rbacEnabled are enabled and namespacePools is disabled
+# 3. Create roles for each namespaces in the namespacePool if enabled.
```

Comment:
I think it might be good to structure our tests around these three different scenarios, since they're so complicated, and hard to match which test case is mapping up to which entry in this list.

Reply:
I didn't really know how to approach testing on these fields:
- Test the RBAC field with these 3 distinct scenarios 
- Or Test the whole feature, like `namespacePools` and all resources impacted by this feature.

In the end I decided to go with the second option.
- First and Second scenario are tested in a commander related tests.
- Third scenario is tested in the namespace pools feature test file

If you want me to change it I can do it, but I believed grouping tests with other ones related functional-wise was better, as long as all test scenarios are covered

Reply:
My original comment was mostly from the difficulty I had reading the tests and knowing which configuration was being tested. So, even if the tests passed, I had a hard time discerning if the tests were valid and valuable. If anything, we should add more comments about what is being tested, or break up `test_astronomer_commander_rbac_cluster_role` into test cases for specific configurations and add docstrings for those.

Reply:
Oh I understand, I'll split the tests to make them easier to understand